client_perms/client.sh

```diff
@@ -0,0 +1,208 @@
+#!/bin/sh
+#
+# Environment variables used:
+#  - SERVER: hostname or IP-address of the NFS-server
+#  - EXPORT: NFS-export to test (should start with "/")
+
+# enable some more output
+set -x
+
+[ -n "${SERVER}" ]
+[ -n "${EXPORT}" ]
+
+if [ "$1" = "client_initialization" ]
+then
+	# install build and runtime dependencies
+	yum -y install nfs-utils time
+
+	mkdir -p /mnt/ganesha
+
+	mount -t nfs -o vers=3 ${SERVER}:${EXPORT} /mnt/ganesha
+
+	echo "Client Initial Stage --- With All Rights To All Clients ( RO & RW ) "
+
+	cd /mnt/ganesha
+	
+	echo "Trying To Write A File"
+	echo "Hello World" > testFile.txt
+	ret=$?
+	if [ $ret -eq 0 ]
+	then
+		echo "SUCCESS"
+	else
+		echo "FAILED ON WRITING RIGHTS"
+		#exit ret
+	fi
+
+	echo "Trying To Read A File"
+	cat testFile.txt
+	ret=$?
+	if [ $ret -eq 0 ]
+	then
+		echo "SUCCESS"
+	else
+		echo "FAILED ON READING RIGHTS"
+		#exit ret
+	fi
+
+	cd / && umount /mnt/ganesha
+fi
+
+if [ "$1" = "client_stage1" ]
+then
+	echo "Client Stage 1 --- With Only RO Rights To This Client "
+
+	mount -t nfs -o vers=3 ${SERVER}:${EXPORT} /mnt/ganesha
+
+	cd /mnt/ganesha
+
+	echo "Trying To Write A File"
+	sed -i '1s/$/ From RedHat/' testFile.txt
+	ret=$?
+	if [ $ret -eq 0 ]
+	then
+		echo "FAILURE Since Write Permissions Were Not Blocked To The Client"
+		#exit ret
+	else
+		echo "SUCCESS ON WRITE PERMISSIONS FAILURE"
```

Comment:
If output is parsed, this says FAILURE, and so will probably trigger a failure.  Either way, this message can probably be cleaned up.

Reply:
The output is not parsed instead the return code is checked ....